File: docs/algorithms/code/algorithms/sortedSquares.ts
function sortedSquares(nums: number[]): number[] {
  nums.sort((a, b) => Math.abs(a) - Math.abs(b));
  return nums.map((n) => n * n);
}

function sortedSquares(nums: number[]): number[] {
  const result: number[] = [];

  for (let i = 0, j = nums.length - 1, pos = nums.length - 1; i <= j; pos--) {
    if (nums[i] * nums[i] > nums[j] * nums[j]) {
      result[pos] = nums[i] * nums[i];
      i++;
    } else {
      result[pos] = nums[j] * nums[j];
      j--;
    }
  }

  return result;
}

File: docs/algorithms/code/algorithms/sortedSquares.go
func sortedSquares(nums []int) []int {

	res := make([]int, len(nums))

	left, right := 0, len(nums)-1
	pos := len(nums) - 1

	for left <= right {
		leftVal := nums[left] * nums[left]
		rightVal := nums[right] * nums[right]

		if leftVal >= rightVal {
			res[pos] = leftVal
			left++
		} else {
			res[pos] = rightVal
			right--
		}
		pos--
	}

	return res
}